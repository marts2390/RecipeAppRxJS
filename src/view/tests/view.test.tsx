import React from 'react'
import { shallow } from 'enzyme'
import View from '../component'

let wrapper
let props

beforeEach(() => {
  props = {
    initialLoad: jest.fn(),
    isLoaded: false,
  }
  wrapper = shallow(<View { ...props } />)
})

test('should render loader', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should render router', () => {
  wrapper.setProps({ isLoaded: true })
  expect(wrapper).toMatchSnapshot()
})

test('get initial load call', () => {
  const instance = wrapper.instance()
  instance.componentDidMount()
  expect(props.initialLoad).toHaveBeenCalled()
})

