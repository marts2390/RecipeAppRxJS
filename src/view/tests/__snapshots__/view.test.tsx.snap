// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render loader 1`] = `
<div
  className="appScreen"
>
  <div
    className="spinnerContainer"
  >
    <LoaderComponent />
  </div>
</div>
`;

exports[`should render router 1`] = `
<div
  className="appScreen"
>
  <GeneralRouterComponent />
</div>
`;
